core/src/main/scala/kafka/server/KafkaConfig.scala

```diff
@@ -1665,6 +1665,7 @@ class KafkaConfig private(doLog: Boolean, val props: java.util.Map[_, _], dynami
   def usesSelfManagedQuorum: Boolean = processRoles.nonEmpty
 
   val migrationEnabled: Boolean = getBoolean(KafkaConfig.MigrationEnabledProp)
+  def enableZkApiForwarding: Boolean = migrationEnabled && interBrokerProtocolVersion.isApiForwardingSupported
```

Comment:
Good question @dengziming. We said in the KIP that forwarding is enabled in all cases with 3.4-IV0, though technically we only need to enable it if we're doing a migration. It might be safer to keep forwarding off when we haven't enabled the migration. Since we won't start the migration until all brokers come up with `zookeeper.metadata.migration.enable`, I think leaving it off like this will be okay.